### ISSUE TITLE
Externally actor.close will never interrupt actor if he wait on a future

### ISSUE THREAD

**Issue by Zelldon** [zeebe-io/zeebe#704]
Example the actor will never closed until the waiting future is completed.
```java
    public void shouldActorCloseExternallyEvenIfRunOnCompletion() throws Exception
    {
        // given
        final CountDownLatch latch = new CountDownLatch(1);
        final CompletableActorFuture<Void> waitingFuture = new CompletableActorFuture<>();
        final ZbActor actor = new ZbActor()
        {
            @Override
            protected void onActorStarted()
            {
                actor.runOnCompletion(waitingFuture, (v, t) -> { });
            }

            @Override
            protected void onActorClosing()
            {
                latch.countDown();
            }
        };
        final ActorFuture<Void> startingFuture = schedulerRule.submitActor(actor);
        startingFuture.join();

        // when
        final Future<Void> future = actor.actor.close();

        // then
        future.get(5, TimeUnit.MINUTES);
        if (!latch.await(5, TimeUnit.SECONDS))
        {
            fail();
        }

    }
```

What we expect ? We should discuss this.

**Comment by Zelldon** [comment]
@menski we decided that calling close should also force to stop waiting on futures, which are submitted at the Started phase, right?

**Comment by menski** [comment]
@Zelldon not sure if I remember correctly what you discussed. Maybe @ThorbenLindhauer or @meyerdan remembers?

**Comment by menski** [comment]
We decided that we want to build and API which allows the developer to specify if we wait for a future before closing

**Comment by Zelldon** [comment]
@ThorbenLindhauer where in the client we need the blocking method?
In SubscriberGroup#openSubscriber?

**Comment by ThorbenLindhauer** [comment]
Review:

* Fix test in https://github.com/zeebe-io/zb-util/commit/e9583ce6c77028441f711325fdce95e97f6de1c1
* I find the implementation indirect and not so easy to follow. Maybe the following is more straightforward:
  1. When creating an `ActorFutureSubscription`, store two things: The phases it triggers in (bitmask) and if the actor should block in those phases until the subscription triggers.
  1. Extend `Subscription` interface by `isBlocking` method. Default impl always returns `false`. `ActorFutureSubscription` impl returns configured value
  1. In `ActorTask#allPhaseSubscriptionsTriggered` (which is the guarding condition for a state change), check if there are any blocking subscriptions that must trigger in the current phase. If yes, then do not make the state change.
* `ActorLifecyclePhasesAndBlockPhaseTest#shouldWaitOnFutureSubmittedInCallback` is a duplicate of `ActorLifecyclePhasesAndRunOnCompletionTest#shouldWaitOnFutureSubmittedInCallback`
